sources/OpenMcdf.Extensions/OLEProperties/OLEProperty.cs

```diff
@@ -33,10 +33,30 @@ public VTPropertyType VTType
             internal set;
         }
 
+        object value;
+
         public object Value
         {
-            get;
-            set;
+            get
+            {
+                switch (VTType)
+                {
+                    case VTPropertyType.VT_LPSTR:
+                    case VTPropertyType.VT_LPWSTR:
+                        if (value is string str && !String.IsNullOrEmpty(str))
+                            return str.Trim('\0');
```

Comment:
Should it be using ```Trim``` or ```TrimEnd```?

Reply:
In my interpretation of specs, it sounds better as a fulltrim. Only embedded nulls are preserved in the presentation. Any thought on this? Thanks

Reply:
The documentation for ```CodePageString``` says
```
the manner in which strings with embedded or additional trailing null characters are presented by the implementation to an application is implementation-specific
```
Which sounds like you can take whichever approach you want, and I do like the idea of being able to see as much of the data as possible.

*However*, for comparison, it does look to me like if I use OpenMcdf to write a user property with leading and trailing nulls then niether Windows Explorer nor Word will display the rest of the data:
![image](https://github.com/ironfede/openmcdf/assets/1178570/b4698b8f-815f-468f-9326-800d01331a94)

Reply:
Is anyone worried about performance for the check and trim on every call?

Reply:
> In my interpretation of specs, it sounds better as a fulltrim. Only embedded nulls are preserved in the presentation. Any thought on this? Thanks

https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-oleps/9660cb24-953a-4e60-adf2-37cc0e779d19

> The string represented by this field SHOULD NOT contain embedded or additional trailing null characters.

So, I think you're safe with `TrimEnd`